### PR TITLE
feat: check environment variables are set before services start

### PR DESCRIPTION
**Pull request by Hyper3x**
Added envalid to **metrics**, **workflow**, **search** and **documents** for strict environment variable checks.

### CODE REVIEW

File: packages/metrics/src/features/metrics/constants.ts [hunk 1]
```diff
@@ -8,13 +8,47 @@
  *
  * Copyright (C) The OpenCRVS Authors located at https://github.com/opencrvs/opencrvs-core/blob/master/AUTHORS.
  */
-export const OPENCRVS_SPECIFICATION_URL = 'http://opencrvs.org/specs/'
-export const CRUD_BIRTH_RATE_SEC = 'id/statistics-crude-birth-rates'
-export const JURISDICTION_TYPE_SEC = 'id/jurisdiction-type'
-export const TOTAL_POPULATION_SEC = 'id/statistics-total-populations'
-export const MALE_POPULATION_SEC = 'id/statistics-male-populations'
-export const FEMALE_POPULATION_SEC = 'id/statistics-female-populations'
-export const JURISDICTION_TYPE_IDENTIFIER = 'id/jurisdiction-type'
```
Comment (naftis) [inline packages/metrics/src/features/metrics/constants.ts:17]: Hey sorry, you can keep these values as constants. They are not going to be configured. Anything starting with process.env is more important

Comment (naftis) [reply packages/metrics/src/features/metrics/constants.ts:17]: In this file in metrics there's actually nothing to configure

File: packages/search/src/environment.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  PORT: port({ devDefault: 9090 }),
+  ES_HOST: str({ devDefault: 'localhost:9200' }),
+  FHIR_URL: url({ devDefault: 'http://localhost:3447/fhir' }),
+  CERT_PUBLIC_KEY_PATH: str({ devDefault: '../../.secrets/public-key.pem' }),
+  USER_MANAGEMENT_URL: url({ devDefault: 'http://localhost:3030/' }),
+  MATCH_SCORE_THRESHOLD: num({ default: 1.0 }),
+  OPENCRVS_INDEX_NAME: str({ devDefault: 'ocrvs' }),
+  DEFAULT_TIMEOUT: num({ devDefault: 600000 }),
+  APPLICATION_CONFIG_URL: url({ devDefault: 'http://localhost:2021/' }),
+  HEARTH_MONGO_URL: str({ devDefault: 'mongodb://localhost/hearth-dev' })
```
Comment (naftis) [inline packages/search/src/environment.ts:14]:
```suggestion
  HOST: str({ default: '0.0.0.0' }),
  PORT: port({ default: 9090 }),
  ES_HOST: str({ default: 'localhost:9200' }),
  FHIR_URL: url({ default: 'http://localhost:3447/fhir' }),
  CERT_PUBLIC_KEY_PATH: str({ default: '../../.secrets/public-key.pem' }),
  USER_MANAGEMENT_URL: url({ default: 'http://localhost:3030/' }),
  MATCH_SCORE_THRESHOLD: num({ default: 1.0 }),
  OPENCRVS_INDEX_NAME: str({ default: 'ocrvs' }),
  DEFAULT_TIMEOUT: num({ default: 600000 }),
  APPLICATION_CONFIG_URL: url({ default: 'http://localhost:2021/' }),
  HEARTH_MONGO_URL: str({ default: 'mongodb://localhost/hearth-dev' })
```

Also the default timeout can be removed and used as a constant string similarly as the metrics comment 👍

File: packages/documents/src/constants.ts [hunk 3]
```diff
@@ -8,11 +8,18 @@
...
+  CERT_PUBLIC_KEY_PATH: str({ default: '../../.secrets/public-key.pem' }),
+  DEFAULT_TIMEOUT: num({ default: 600000 })
+})
+
+export const HOST = env.DOCUMENTS_HOST
+export const PORT = env.DOCUMENTS_PORT
+export const CERT_PUBLIC_KEY_PATH = env.CERT_PUBLIC_KEY_PATH
 export const SENTRY_DSN = process.env.SENTRY_DSN
-export const DEFAULT_TIMEOUT = 600000
+export const DEFAULT_TIMEOUT = env.DEFAULT_TIMEOUT
```
Comment (naftis) [inline packages/documents/src/constants.ts:24]: You can revert this change

File: packages/workflow/src/environment.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  WEBHOOKS_URL: url({ devDefault: 'http://localhost:2525/' }),
+  METRICS_URL: url({ devDefault: 'http://localhost:1050/' }),
+  COUNTRY_CONFIG_URL: url({ devDefault: 'http://localhost:3040' }),
+  CERT_PUBLIC_KEY_PATH: str({ devDefault: '../../.secrets/public-key.pem' }),
+  DOCUMENTS_URL: url({ devDefault: 'http://localhost:9050' }),
+  USER_MANAGEMENT_URL: url({ devDefault: 'http://localhost:3030/' }),
+  SENTRY_DSN: str({ default: undefined }),
+  AUTH_URL: url({ devDefault: 'http://localhost:4040/' }),
+  DEFAULT_TIMEOUT: num({ devDefault: 600000 }),
+  LANGUAGES: str({ devDefault: 'en,fr' })
```
Comment (naftis) [inline packages/workflow/src/environment.ts:19]:
```suggestion
  HOST: str({ default: '0.0.0.0' }),
  PORT: num({ default: 5050 }),
  FHIR_URL: url({ devDefault: 'http://localhost:3447/fhir' }),
  APPLICATION_CONFIG_URL: url({ devDefault: 'http://localhost:2021/' }),
  NOTIFICATION_SERVICE_URL: url({ devDefault: 'http://localhost:2020/' }),
  SEARCH_URL: url({ devDefault: 'http://localhost:9090/' }),
  WEBHOOKS_URL: url({ devDefault: 'http://localhost:2525/' }),
  METRICS_URL: url({ devDefault: 'http://localhost:1050/' }),
  COUNTRY_CONFIG_URL: url({ devDefault: 'http://localhost:3040' }),
  CERT_PUBLIC_KEY_PATH: str({ default: '../../.secrets/public-key.pem' }),
  DOCUMENTS_URL: url({ devDefault: 'http://localhost:9050' }),
  USER_MANAGEMENT_URL: url({ devDefault: 'http://localhost:3030/' }),
  SENTRY_DSN: str({ default: undefined }),
  AUTH_URL: url({ devDefault: 'http://localhost:4040/' }),
  DEFAULT_TIMEOUT: num({ devDefault: 600000 }),
  LANGUAGES: str({ default: 'en,fr' })
```

Also can revert the DEFAULT_TIMEOUT to const

**Review by naftis — CHANGES_REQUESTED** [review]
Minor changes, otherwise looks good! Thank you :)

File: packages/workflow/src/environment.ts (hunk 4 above)
Comment (Hyper3x) [reply packages/workflow/src/environment.ts]: ok thank you @naftis, I fixed the issues.
